lib/autoupdate.ps1

```diff
@@ -351,77 +398,23 @@ function get_version_substitutions([String] $version, [Hashtable] $customMatches
 
 function autoupdate([String] $app, $dir, $json, [String] $version, [Hashtable] $matches) {
     Write-Host -f DarkCyan "Autoupdating $app"
-    $has_changes = $false
-    $has_errors = $false
-    [Bool]$valid = $true
     $substitutions = get_version_substitutions $version $matches
 
-    if ($json.url) {
-        # create new url
-        $url   = substitute $json.autoupdate.url $substitutions
-        $valid = $true
-
-        if($valid) {
-            # create hash
-            $hash = get_hash_for_app $app $json.autoupdate.hash $version $url $substitutions
-            if ($null -eq $hash) {
-                $valid = $false
-                Write-Host -f DarkRed "Could not find hash!"
-            }
-        }
-
-        # write changes to the json object
-        if ($valid) {
-            $has_changes = $true
-            update_manifest_with_new_version $json $version $url $hash
-        } else {
-            $has_errors = $true
-            throw "Could not update $app"
-        }
-    } else {
-        $json.architecture | Get-Member -MemberType NoteProperty | ForEach-Object {
-            $valid = $true
-            $architecture = $_.Name
-
-            # create new url
-            $url   = substitute (arch_specific "url" $json.autoupdate $architecture) $substitutions
-            $valid = $true
-
-            if($valid) {
-                # create hash
-                $hash = get_hash_for_app $app (arch_specific "hash" $json.autoupdate $architecture) $version $url $substitutions
-                if ($null -eq $hash) {
-                    $valid = $false
-                    Write-Host -f DarkRed "Could not find hash!"
-                }
-            }
-
-            # write changes to the json object
-            if ($valid) {
-                $has_changes = $true
-                update_manifest_with_new_version $json $version $url $hash $architecture
-            } else {
-                $has_errors = $true
-                throw "Could not update $app $architecture"
-            }
-        }
-    }
-
     # update properties
-    update_manifest_prop "extract_dir" $json $substitutions
-
-    # update license
-    update_manifest_prop "license" $json $substitutions
+    $updatedProperties = @(@($json.autoupdate.PSObject.Properties.Name) -ne 'architecture')
+    if ($json.autoupdate.architecture) {
+        $updatedProperties += $json.autoupdate.architecture.PSObject.Properties | ForEach-Object { $_.Value.PSObject.Properties.Name } | Select-Object -Unique
+    }
+    if (!($updatedProperties -eq 'hash')) {
```

Comment:
This should be array all the time. (L404)

```suggestion
    if ($updatedProperties -notcontains 'hash') {
```

Reply:
The following is more precise:

```powershell
    if ($updatedProperties -contains 'url') {
        $updatedProperties += 'hash'
    }
    $updatedProperties = $updatedProperties | Select-Object -Unique
```

But I don't think there will be `autoupdate` without `au.url`... So `$updatedProperties -notcontains 'hash'` should be okay.